packages/gravity/blade-runner/src/plan/run-process.ts

```diff
@@ -6,154 +6,124 @@ import { mkdir, readFile } from 'node:fs/promises';
 import { createServer } from 'node:http';
 import { type AddressInfo } from 'node:net';
 import { join } from 'node:path';
-import type { Browser, BrowserType } from 'playwright';
-import { v4 } from 'uuid';
+import type { BrowserContext, BrowserType } from 'playwright';
 
-import { Trigger, scheduleMicroTask } from '@dxos/async';
 import { Context } from '@dxos/context';
 import { invariant } from '@dxos/invariant';
 import { type LogProcessor, log, createFileProcessor, LogLevel, CONSOLE_PROCESSOR } from '@dxos/log';
 
-import { type AgentResult, type AgentParams, type PlanOptions, type Platform, AGENT_LOG_FILE } from './spec';
+import { type ReplicantParams, type GlobalOptions, type Platform, type ReplicantRuntimeParams } from './spec';
 
 const DEBUG_PORT_START = 9229;
 
-type ProcessHandle = {
+export type ProcessHandle = {
   /**
-   * Promise that resolves when the agent finishes.
+   * Kill the replicant process/browser.
    */
-  result: Promise<AgentResult>;
+  kill: (signal?: NodeJS.Signals | number) => void;
+};
 
-  /**
-   * Kill the agent process/browser.
-   */
-  kill: () => void;
+export type RunParams<S> = {
+  agentParams: ReplicantParams<S>;
+  options: GlobalOptions;
 };
 
-export const runNode = <S, C>(
-  planName: string,
-  agentParams: AgentParams<S, C>,
-  options: PlanOptions,
-): ProcessHandle => {
+export const runNode = <S>(params: RunParams<S>): ProcessHandle => {
   const execArgv = process.execArgv;
 
-  if (options.profile) {
+  if (params.options.profile) {
     execArgv.push(
       '--cpu-prof', //
       '--cpu-prof-dir',
-      agentParams.outDir,
+      params.agentParams.outDir,
       '--cpu-prof-name',
       'agent.cpuprofile',
     );
   }
 
-  const startTs = performance.now();
   const childProcess = fork(process.argv[1], {
-    execArgv: options.debug
+    execArgv: params.options.debug
       ? [
-          '--inspect=:' + (DEBUG_PORT_START + agentParams.agentIdx), //
+          '--inspect=:' + (DEBUG_PORT_START + params.agentParams.replicantId), //
           ...execArgv,
         ]
       : execArgv,
     env: {
       ...process.env,
-      GRAVITY_AGENT_PARAMS: JSON.stringify(agentParams),
-      GRAVITY_SPEC: planName,
+      DX_RUN_PARAMS: JSON.stringify(params),
```

Comment:
can those two env variables be combined into one?

Reply:
Yes

Reply:
But I prefer to keep them separate for code clarity in `main.ts`